### PR TITLE
Ported the ProfiledPIDSubsystem from the wpilib java source to Python

### PR DESCRIPTION
**Pull request by cwstryker**
Used ChatGPT to get 80% of the job done.  This is the ProfiledPIDSubsystem only.  No new tests were added.  I verified functionality using a separate robot simulation project.

### CODE REVIEW

File: commands2/profiledpidsubsystem.py [hunk 1]
```diff
@@ -0,0 +1,62 @@
+# Copyright (c) FIRST and other WPILib contributors.
+# Open Source Software; you can modify and/or share it under the terms of
+# the WPILib BSD license file in the root directory of this project.
+
+from wpimath.controller import ProfiledPIDController
+from .subsystem import Subsystem
+from wpimath.trajectory import TrapezoidProfile
+
+
+class ProfiledPIDSubsystem(Subsystem):
```
Comment (virtuald) [inline commands2/profiledpidsubsystem.py:10]: Comments missing for class

File: commands2/profiledpidsubsystem.py [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+    def __init__(self, controller: ProfiledPIDController, initial_position: float = 0):
+        super().__init__()
+        self._m_controller = controller
```
Comment (virtuald) [inline commands2/profiledpidsubsystem.py:13]: Get rid of the `_m`

File: commands2/profiledpidsubsystem.py [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+        self._m_enabled = False
+        self.setGoal(initial_position)
+
+    def periodic(self):
+        """Updates the output of the controller."""
+        if self._m_enabled:
+            self._useOutput(self._m_controller.calculate(self._getMeasurement()),
+                            self._m_controller.getSetpoint())
+
+    def get_controller(self) -> ProfiledPIDController:
```
Comment (virtuald) [inline commands2/profiledpidsubsystem.py:23]: `getController`

**Review by virtuald — CHANGES_REQUESTED** [review]
Mostly good, thanks

File: commands2/__init__.py [hunk 4]
```diff
@@ -33,6 +33,7 @@
 from .pidcommand import PIDCommand
 from .pidsubsystem import PIDSubsystem
 from .printcommand import PrintCommand
+from .profiledpidsubsystem import ProfiledPIDSubsystem
```
Comment (virtuald) [inline commands2/__init__.py:36]: This needs to be in `__all__`

File: commands2/profiledpidsubsystem.py (hunk 1 above)
Comment (cwstryker) [reply commands2/profiledpidsubsystem.py:10]: Added a docstring.

File: commands2/profiledpidsubsystem.py (hunk 2 above)
Comment (cwstryker) [reply commands2/profiledpidsubsystem.py]: Done.

File: commands2/profiledpidsubsystem.py (hunk 3 above)
Comment (cwstryker) [reply commands2/profiledpidsubsystem.py]: Done.

File: commands2/__init__.py (hunk 4 above)
Comment (cwstryker) [reply commands2/__init__.py:36]: Done.

**Review by cwstryker — COMMENTED** [review]
I think I fixed all of the issues.

**Comment by cwstryker** [comment]
I see that some of the checks are failing.  I think I understand the cause of the issue.  I will work on fixing it tomorrow.

**Comment by virtuald** [comment]
Thanks, everything looks good.

There are multiple ProfiledPIDControllers (one for dimensionless, and one for radians), and even though they're effectively the same thing there are some C++ classes that require one or the other. If you have any experience with python's typing generics, any ideas you have to make this work with both types of ProfiledPIDControllers would be welcomed.

I posted some thoughts on a related issue at https://github.com/robotpy/robotpy-commands-v2/pull/40, but I'm still mulling it over. 

If you don't have any ideas or expertise in this, that's fine. I'll contemplate and merge/adjust this once #46 is finished, which hopefully will be soon.

**Comment by virtuald** [comment]
I'm going to merge this without typing for now. I don't really like the union approach, and I'm too tired to think through the generic/typevar approach. We can add types back at a later date.

**Pull request merged**